generator/src/lib.rs

```diff
@@ -1523,21 +1538,62 @@ pub fn derive_setters(_struct: &vkxml::Struct) -> Option<Tokens> {
         })
     });
 
+    let mut nexts = Vec::new();    
+    let name_builder_next = name_to_tokens(&(_struct.name.clone() + "BuilderNext"));        
+    if let Some(extends) = &_struct.extends {
+        for target in extends.split(',') {
+            let target_ident = name_to_tokens(&(target.to_string() + "BuilderNext"));
+            nexts.push(quote! {
+                impl #target_ident for #name_builder_next {}
+            });
+        }
+    }
+
+    let next_function = if has_next {
+        if is_next_const {
+            quote!{
+                pub fn next<T>(mut self, next: &T) -> #name_builder<'a> where T: #name_builder_next {
+                    self.inner.p_next = next as *const T as *const c_void;
+                    self
+                }
+            }
+        } else {
+            quote!{
+                pub fn next<T>(mut self, next: &mut T) -> #name_builder<'a> where T: #name_builder_next {
+                    self.inner.p_next = next as *mut T as *mut c_void;
+                    self
+                }
+            }
+        }
+    } else {
+        quote!{}
+    };
+
+    let next_trait = if has_next {
+        quote!{pub trait #name_builder_next {}}
```

Comment:
```suggestion
        quote!{pub unsafe trait #name_builder_next {}}
```

Reply:
Up to this day (and I wrote about 20 000 lines of code in Rust for my bachelor thesis :D) I didn't know you could also mark a trait as unsafe. I leave this decision to @MaikKlein while I google away to learn about unsafe trait.

Reply:
I don't even think that a user would want to implement those traits manually, but if they do they better make sure it is valid, so yes it has to be unsafe.

Should this trait even be public?

Reply:
I agree with the indication of the fact that it is unsafe to implement it.

In theory it shouldn't be public but I can see that someone may be stuck for some reason on older Ash version in future but may want to add an extension from newer Vulkan.

Reply:
There's at least theoretically a population users of unpublished extensions, too.